phase/configure_k0s.go

```diff
@@ -309,6 +309,10 @@ func (p *ConfigureK0s) configFor(h *cluster.Host) (string, error) {
 	cfg.DigMapping("spec", "api")["address"] = addr
 	addUnlessExist(&sans, addr)
 
+	if externalAddr := cfg.DigString("spec", "api", "externalAddress"); externalAddr != "" {
+		addUnlessExist(&sans, externalAddr)
```

Comment:
Should externalAddr be in sans?

Reply:
No, it's added automatically. the component `endpoint-reconciler` should be disabled though. Everything else seems fine

Reply:
I don't think ctl needs to worry about those. Maybe all the other sans manipulation is pointless too if k0s does it on its own?

Reply:
```go
// Sans return the given SANS plus all local addresses and externalAddress if given
func (a *APISpec) Sans() []string {
	sans, _ := iface.AllAddresses()
	sans = append(sans, a.Address)
	sans = append(sans, a.SANs...)
	if a.ExternalAddress != "" {
		sans = append(sans, a.ExternalAddress)
	}

	return stringslice.Unique(sans)
}
```

k0s does all this, I don't think ctl needs to manipulate sans at all.

Reply:
ctl added the FQDNs to the SANs (see https://github.com/k0sproject/k0s/issues/5168).